### ISSUE TITLE
inference-server  failed when docker compose up 

### ISSUE THREAD

**Issue by is** [LAION-AI/Open-Assistant#2980]
```
docker compose --profile inference up --attach-dependencies --force-recreate
```
I got the error info
```
open-assistant-inference-server-1  | ERROR:    Traceback (most recent call last):
open-assistant-inference-server-1  |   File "/var/opt/inference/server/lib/starlette/routing.py", line 671, in lifespan
open-assistant-inference-server-1  |     async with self.lifespan_context(app):
open-assistant-inference-server-1  |   File "/var/opt/inference/server/lib/starlette/routing.py", line 566, in __aenter__
open-assistant-inference-server-1  |     await self._router.startup()
open-assistant-inference-server-1  |   File "/var/opt/inference/server/lib/starlette/routing.py", line 648, in startup
open-assistant-inference-server-1  |     await handler()
open-assistant-inference-server-1  |   File "/opt/inference/server/main.py", line 44, in enable_prom_metrics
open-assistant-inference-server-1  |     Instrumentator().instrument(app).expose(app)
open-assistant-inference-server-1  |   File "/var/opt/inference/server/lib/prometheus_fastapi_instrumentator/instrumentation.py", line 146, in __init__
open-assistant-inference-server-1  |     raise ValueError(
open-assistant-inference-server-1  | ValueError: Env var PROMETHEUS_MULTIPROC_DIR='/tmp/oasst-inference-prometheus-metrics' not a directory.
```

I think that the error is related by inference/server/server_main.sh
```
gunicorn_workers=${GUNICORN_WORKERS:-1}

# if 1 worker, stay with uvicorn
if [ $gunicorn_workers -eq 1 ]; then
    unset PROMETHEUS_MULTIPROC_DIR
    uvicorn main:app --host 0.0.0.0 --port "$PORT"
else
    if [ -d "${PROMETHEUS_MULTIPROC_DIR}" ]; then rm -rf "${PROMETHEUS_MULTIPROC_DIR}"; fi
    mkdir -p "${PROMETHEUS_MULTIPROC_DIR}"
    port=${PORT:-8080}
    gunicorn main:app --workers $gunicorn_workers --worker-class uvicorn.workers.UvicornWorker --bind 0.0.0.0:$port
fi

```
and 
```
# add prometheus metrics at /metrics
@app.on_event("startup")
async def enable_prom_metrics():
    Instrumentator().instrument(app).expose(app)
```
in inference/server/main.py.

I don't quite understand the detailed mechanism, so I don't know how to fix it.

**Comment by is** [comment]
Commenting out the hook directly can temporarily solve the problem, at least there won't be any error messages.

```
index 599fec0..dcf650e 100644
--- a/inference/server/main.py
+++ b/inference/server/main.py
@@ -39,9 +39,9 @@ async def log_exceptions(request: fastapi.Request, call_next):
 
 
 # add prometheus metrics at /metrics
-@app.on_event("startup")
-async def enable_prom_metrics():
-    Instrumentator().instrument(app).expose(app)
+#@app.on_event("startup")
+#async def enable_prom_metrics():
+#    Instrumentator().instrument(app).expose(app)
```

**Comment by olliestanley** [comment]
This will be fixed by #2971
